nifpga/session.py

```diff
@@ -390,8 +413,250 @@ def read(self):
+            total_num_bits = self._word_length + overflow_bit
+            combinedData = combinedData >> (32 * self._transfer_len - total_num_bits)
+        return combinedData
+
+    def _convert_from_read_value_to_decimal(self, data):
+        """ This method converts value from hardware and returns the respective
+        decimal value or a tuple with the overflow status and the decimal
+        value.
+        """
+        overflow = None
+        if self._overflow_enabled:
+            overflow = self._get_overflow_value(data)
+            data = self._remove_overflow_bit(data)
+
+        if self._signed:
+            data = self._integer_twos_comp(data)
+        decimal_value = Decimal(data * self._delta)
+        if self._overflow_enabled:
+            return (overflow, decimal_value)
+        else:
+            return decimal_value
+
+    def _get_overflow_value(self, data):
+        """ Mask out all the data within the word length, leaving the overflow
+        bit. If the result after masking the the word portion of the fixed
+        point is none zero that indicates the data read has overflowed. """
+        mask = 2**(self._word_length)
+        if data & mask > 0:
+            return True
+        return False
+
+    def _remove_overflow_bit(self, data):
+        """ This helper method masks out all bits not inside the word length,
+        ultimately returning a value of data without the overflow bit. """
+        return data & (2**(self._word_length) - 1)
+
+    def _integer_twos_comp(self, data):
+        if (data & (1 << (self._word_length - 1))) != 0:
+            data = data ^ (2 ** (self._word_length) - 1)
+            data += 1
+            data *= -1
+        return data
+
+    def write(self, user_input):
+        """ Writes the user's the users input into the register as a fixed
+        point number. Any inputs outside the bounds of this fixed point
+        register will be coerced and the user will be warned. The user input's
+        supported include any python Number.
+
+            Args:
+                Number: user numerical input to be converted to fixed point.
+                (bool, Number): Tuple with the members : Boolean for overflow,
+                                user numerical input to be converted to fixed
+                                point.
+        """
+
+        fxp_representation = self._convert_user_input_to_fxp_representation(user_input)
+        arrayData = self._extract_array_of_u32_from_fxp_value(fxp_representation)
+        buf = self._ctype_type(*arrayData)
+        self._write_func(self._session, self._resource, buf, self._transfer_len)
+
+    def _convert_user_input_to_fxp_representation(self, user_input):
+        (overflow, data) = self._validate_and_parse_user_input(user_input)
+
+        fxp_representation = 0
+        if data < self._minimum:
+            fxp_representation = self._convert_value_to_fxp(self._minimum)
+            self.warn_coerced_data()
```

Comment:
It looks like you are double warning on these?  As far as I can tell.  _convert_value_to_fxp already warns.

Reply:
Nevermind I missed that you are passing self._minimum into this.

Reply:
Yeah so the convert_value_to_fxp does not care about minimum or max values/ the warn inside convert warns whenever the input is a non-delta multiple.